ui/src/components/Configuration/ConfigurationDrawer.vue

```diff
@@ -0,0 +1,329 @@
+  activeUpdate: Function,
+  closePanel: { type: Function as PropType<(payload: MouseEvent) => void>, required: true },
+  item: { type: Object as PropType<LocalConfigurationWrapper>, required: true },
+  advancedActive: Boolean,
+  addAdvancedOption: { type: Function as PropType<(payload: MouseEvent) => void>, required: true },
+  advancedKeyUpdate: { type: Function, required: true },
+  deleteAdvancedOption: { type: Function, required: true },
+  saveCurrentState: { type: Function as PropType<(payload: MouseEvent) => void>, required: true },
+  edit: Boolean,
+  helpState: { type: Object, required: true },
+  updateFormValue: { type: Function, required: true },
+  loading: { type: Boolean, required: true }
+})
+
+/**
+ * Local State
+ */
+
+const bounceOutTimeout = ref(-1)
+const bounceInTimeout = ref(-1)
+const initialWatchTimeout = ref(-1)
+
+const configurationDrawerActive = computed(() => props?.configurationDrawerActive)
+const cancelIcon = computed(() => Cancel)
+const helpState = computed(() => props.helpState)
+const editing = computed(() => props.edit)
+const errors = computed(() => props?.item?.errors)
+
+/**
+ * Scrolls the drawer to the first error on creation
+ * (should draw the user's eye to the problem)
+ *  This is unneccessary, but a nice to have.
+ */
+watch(errors, () => {
+  if (props.item?.errors?.hasErrors) {
+    clearTimeout(initialWatchTimeout.value)
+    initialWatchTimeout.value = window.setTimeout(() => {
+      const elem = document.querySelector('.feather-input-error')
+      const wrapper = document.querySelector('.slide-outer-body')
+
+      //Scrolls the drawer to the first error.
+      wrapper?.scrollTo({ top: elem?.getBoundingClientRect().top, behavior: 'smooth' })
+
+      //This 'bounce' animation is an attempt to further draw
+      //the user's eye to the field with the error. It quickly scales and then descales
+      //the element. The delays are to allow the above scroll to complete before moving on.
+      clearTimeout(bounceOutTimeout.value)
+      clearTimeout(bounceInTimeout.value)
+
+      bounceOutTimeout.value = window.setTimeout(() => {
+        const inputWrapper = elem?.parentElement?.parentElement
+        inputWrapper?.classList.add('bounce')
+        inputWrapper?.querySelector<HTMLInputElement>('.feather-input')?.focus()
+        bounceInTimeout.value = window.setTimeout(() => {
+          inputWrapper?.classList.remove('bounce')
+        }, 200)
+      }, 300)
+    }, 50)
+  }
+})
+
+
+/**
+ * Determines which classes to apply to the drawer.
+ */
+const wrapperClass = () => {
+  let classes = 'sidepanelclosed '
+  if (props.configurationDrawerActive) {
+    classes = 'sidepanel '
+  }
+  if (helpState?.value?.open) {
+    classes += 'help-open'
+  }
+  return classes
+}
+
+const toggleHelp = () => {
+  helpState.value.open = !helpState.value.open
+}
+</script>
+
+<style lang="scss">
+.side-label {
+  .group-label {
+    color: var(--feather-primary);
```

Comment:
one more for the cleanup might be to change all these to the recommended way in the feather docs.. eg: `var($primary)`